Source/render/NodeRenderer.swift

```diff
@@ -21,6 +21,29 @@ class NodeRenderer {
     fileprivate var active = false
     weak var animationCache: AnimationCache?
 
+    fileprivate var cachedAbsPlace: Transform?
+    public var absPlace: Transform {
+        get {
+            if let place = cachedAbsPlace {
+                return place
+            }
+
+            if let place = parentRenderer?.absPlace.concat(with: node.place) {
+                cachedAbsPlace = place
+                return place
+            }
+
+            return .identity
+        }
+    }
+    func freeCachedAbsPlace() {
```

Comment:
Did you use swiftlint? It looks like there should be a warning because there is no empty line after previous function.